src/LowestCommonAncestor.php

```diff
@@ -2,35 +2,55 @@
 
 namespace Relaxed\LCA;
 
+use Exception;
 use Fhaculty\Graph\Vertex;
 use Fhaculty\Graph\Graph;
 use Graphp\Algorithms\Search\BreadthFirst;
+use Fhaculty\Graph\Exception\UnderflowException;
 
-class LowestCommonAncestor {
+class LowestCommonAncestor
+{
+    public function __construct(Graph $graph)
+    {
+        $this->graph = $graph;
+    }
 
-  function __construct(Graph $graph) {
-    $this->graph = $graph;
-  }
+    /**
+     * @param Vertex $local
+     * @param Vertex $remote
+     * @param int $direction
+     * Traverse in reverse order the graph when searching by default;
+     * @return bool
+     */
+    public function find(Vertex $local, Vertex $remote, $direction = BreadthFirst::DIRECTION_REVERSE)
```

Comment:
@jeqq @dickolsson , Yes, it won't work same if direction is changed. This will change the complete structure of tree as well. However, I have tested the find() method with DIRECTION_FORWARD and as expected, No LCA was found.